### PR TITLE
Support for bazel builds on macos_arm64

### PR DESCRIPTION
**Pull request by simonmaurer**
with respect to this [issue](https://github.com/google/XNNPACK/issues/1302) this PR allows bazel builds for the recent `macos_arm64` platform.

verified using the following bazel versions:
- bazel-3.7.2-darwin-x86_64
- bazel-4.0.0-darwin-x86_64
- bazel-4.0.0-homebrew
- bazel-3.7.1-darwin-arm64

> bazel -c opt --config=macos_arm64 :end2end_bench

specifying `--config=macos_arm64` is mandatory to allow a succesful build of the `XNNPACK`library
if we wanna build the `ènd2end_bench` another PR has to be made to track the latest commit of `pthreadpool`(once this [PR](https://github.com/Maratyszcza/pthreadpool/pull/14) is accepted)

as [TensorFlow](https://github.com/tensorflow/tensorflow) depends on [XNNPACK](https://github.com/google/XNNPACK), once this PR is accepted we can issue a PR on the official [TensorFlow repo](https://github.com/tensorflow/tensorflow). this will allow bazel builds of the `TensorFlow Lite library` and the `benchmark_model` for platform `macos_arm64`

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -44,3 +44,10 @@ build:ios_arm64e --watchos_cpus=armv7k
 build:ios_fat --config=ios
 build:ios_fat --ios_multi_cpus=armv7,arm64
 build:ios_fat --watchos_cpus=armv7k
+
+# macOS configs.
+build:macos --apple_platform_type=macos
+
+build:macos_arm64 --config=macos
+build:macos_arm64 --cpu=darwin_arm64
+#build:macos_arm64 --noenable_platform_specific_config
```
Comment (Maratyszcza) [inline .bazelrc:53]: No commented out configs please. Either remove it, or uncomment.

File: BUILD.bazel [hunk 2]
```diff
@@ -7482,6 +7482,14 @@ config_setting(
...
 )
 
+config_setting(
+    name = "macos_arm64",
+    values = {
+        "apple_platform_type": "macos",
+        "cpu": "darwin_arm64",
+    },
+)
+
```
Comment (Maratyszcza) [inline BUILD.bazel:7492]: Should be grouped with other macos configs

**Review by Maratyszcza — CHANGES_REQUESTED** [review]

File: .bazelrc (hunk 1 above)
Comment (simonmaurer) [reply .bazelrc]: @Maratyszcza thank you for this note. fixed

File: BUILD.bazel (hunk 2 above)
Comment (simonmaurer) [reply BUILD.bazel]: regrouped with other macos configs.

**Comment by simonmaurer** [comment]
updated pthreadpool library, using latest commit which pulls in [PR #14](https://github.com/Maratyszcza/pthreadpool/pull/14)

`end2end_bench` can now be built using:
> bazel -c opt --config=macos_arm64 :end2end_bench

**Review by Maratyszcza — APPROVED** [review]

**Comment by simonmaurer** [comment]
@Maratyszcza thank you for reviewing!

**Pull request merged**